plugin/trino-exasol/src/test/java/io/trino/plugin/exasol/TestExasolConnectorTest.java

```diff
@@ -0,0 +1,470 @@
+
+            assertThat(query(format("SELECT * FROM %s WHERE c %s %s", table.getName(), operator, filterLiteral)))
+                    .isFullyPushedDown();
+        }
+    }
+
+    @Override
+    @Language("RegExp")
+    protected String errorMessageForInsertIntoNotNullColumn(String columnName)
+    {
+        return format("constraint violation - not null \\(column %s in table.*", columnName.toUpperCase(ENGLISH));
+    }
+
+    @Override
+    @Language("RegExp")
+    protected String errorMessageForCreateTableAsSelectNegativeDate(String date)
+    {
+        return format("negative date %s as select blubb", date);
+    }
+
+    @Override
+    @Language("RegExp")
+    protected String errorMessageForInsertNegativeDate(String date)
+    {
+        return format("negative date %s insert blubb", date);
+    }
+
+    @Override
+    protected void verifyAddNotNullColumnToNonEmptyTableFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageContaining("constraint violation - not null (NN/PK on table");
+    }
+
+    @Override
+    protected void verifyConcurrentAddColumnFailurePermissible(Exception e)
+    {
+        assertThat(e)
+                .hasMessage("xx: The DDL cannot be run concurrently with other DDLs\n");
+    }
+
+    @Override
+    protected void verifySetColumnTypeFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageStartingWith("data exception - ");
+    }
+
+    @Override
+    protected OptionalInt maxSchemaNameLength()
+    {
+        return OptionalInt.of(128);
+    }
+
+    @Override
+    protected void verifySchemaNameLengthFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageContaining("maximum length of identifier exceeded");
+    }
+
+    @Override
+    protected OptionalInt maxTableNameLength()
+    {
+        return OptionalInt.of(128);
+    }
+
+    @Override
+    protected void verifyTableNameLengthFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageStartingWith("maximum length of identifier exceeded");
+    }
+
+    @Override
+    protected OptionalInt maxColumnNameLength()
+    {
+        return OptionalInt.of(128);
+    }
+
+    @Override
+    protected void verifyColumnNameLengthFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageContaining("maximum length of identifier exceeded");
+    }
+
+    @Override
+    protected Optional<String> filterColumnNameTestData(String columnName)
+    {
+        // Exasol does not support '.' in identifiers
+        if (columnName.contains(".")) {
+            return Optional.empty();
+        }
+        return Optional.of(columnName);
+    }
+
+    @Override
+    @Test
+    public void testDeleteWithVarcharGreaterAndLowerPredicate()
+    {
+        abort("Exasol does not support DELETE with < or > predicate");
+    }
+
+    @Override
+    @Test
+    public void testDeleteWithVarcharInequalityPredicate()
+    {
+        abort("Exasol does not support DELETE with != predicate");
+    }
```

Comment:
can probably return `false` for `SUPPORTS_ROW_LEVEL_DELETE` from `hasBehaviour`.